packages/blockchain-tools/src/interfaces/StarknetWallet.ts

```diff
@@ -49,182 +45,198 @@ class StarknetWallet extends ChainBaseClass {
   }
 
   async getWallet() {
-    const wallet = await connect({
-      modalMode: "alwaysAsk",
-    })
-    console.log("Starknet Wallet", wallet)
-    return wallet
+    const { wallet, connector } = await connect({
```

Comment:
I think this getWallet function is doing what the `connect` function is supposed to do. Can the logic be moved into `connect`? You might have to expand the return type to include an optional `connector` type.